### ISSUE TITLE
Diff of two CellML files

### ISSUE THREAD

**Issue by agarny** [opencor/opencor#25]
This is a feature on which many software developers rely, and which would be equally useful to modellers and model users.

**Comment by nickerso** [comment]
Just reviving this issue following a mention on #1212 - this would be a useful feature when looking at changes between revisions of a model or simulation description when staging changes for synchronising to PMR.

I don't think it would be useful for OpenCOR users to be doing a simple line-based diff on CellML XML files as per a standard Git client. Some ideas were kicked around here in Auckland, mostly with myself and @dbrnz, so I thought it would be useful to list them here.

* a diff of the CellML text view of the revisions of the model
* the new model comparison features in PMR could potentially be used as a webservice and presented in the web widget inside OpenCOR
  * these use the SEMS tools mentioned above and PMR could provide that wrapper to make them accessible in OpenCOR.
  * provides visualisation of differences in the *model* (or SED-ML, maybe) rather than the XML syntactical differences
  * does provide a serialisation of the differences which could maybe be used to highlight things in the existing OpenCOR views?
  * would be good if these could work on local revisions rather than needing things to be first pushed to PMR

Not in any way suggesting these as high priority features for either OpenCOR or PMR!

**Comment by agarny** [comment]
At this stage, I would favour the first option (i.e. do a diff of the CellML Text view). Although we clearly ought to try your other suggestion, I fear that it might be unnecessarily slow (esp. from Europe) while I would expect such a feature to be fast. Anyway, not our top priority, I agree.

**Comment by mirams** [comment]
We've been using BiVeS on the Web Lab. Really good for a human readable diff: https://travis.cs.ox.ac.uk/FunctionalCuration/compare/m/90/89/show/-1805825569/displayBivesDiff

If a little slow to render...

**Comment by nickerso** [comment]
yep - that is what the *compare* tool in PMR uses. Its a nice way to easily see the differences between versions of a model.
